ksqldb-engine/src/main/java/io/confluent/ksql/internal/PullQueryExecutorMetrics.java

```diff
@@ -130,9 +131,10 @@ public void recordLatency(
       final long startTimeNanos,
       final PullSourceType sourceType,
       final PullPhysicalPlanType planType,
-      final RoutingNodeType routingNodeType
+      final RoutingNodeType routingNodeType,
+      final DataSourceType dataSourceType
   ) {
-    final MetricsKey key = new MetricsKey(sourceType, planType, routingNodeType);
+    final MetricsKey key = new MetricsKey(sourceType, planType, routingNodeType, dataSourceType);
```

Comment:
Here's where we're actually adding the new type tag. Question: is this a compatible change?